api/nodes/serializers.py

```diff
@@ -130,6 +131,128 @@ def update(self, instance, validated_data):
+    date_registered = ser.DateTimeField(read_only=True)
+    gravatar_url = ser.CharField(read_only=True, help_text='URL for the icon used to identify the user. '
+                                                           'Relies on http://gravatar.com ')
+    employment_institutions = ser.ListField(read_only=True, source='jobs', help_text='An array of dictionaries '
+                                                                                     'representing the places the '
+                                                                                     'user has worked')
+    educational_institutions = ser.ListField(read_only=True, source='schools',
+                                             help_text='An array of dictionaries representing the places the '
+                                                       'user has attended school')
+    social_accounts = ser.DictField(read_only=True, source='social',
+                                    help_text='A dictionary of various social media account dentifiers including '
+                                              'an array of user-defined URLs')
+    bibliographic = ser.BooleanField(help_text='Whether the user will be included in citations for this node or not.  '
+                                               'Required due to issues with field defaulting to false.')
+
+    permission = ser.ChoiceField(choices=['read', 'write', 'admin'], required=False, allow_null=True,
+                                 help_text='Highest permission the user has.  Blank input defaults write permission if '
+                                           'user is being added and to current permission if user is being edited.')
+    links = LinksField({
+        'html': 'absolute_url',
+        'nodes': {
+            'relation': Link('users:user-nodes', kwargs={'user_id': '<pk>'})
+        },
+        'detail': Link('nodes:node-contributor-detail',
+                       kwargs={'node_id': '<node_id>', 'user_id': '<pk>'})
+    })
+
+    class Meta:
+        type_ = 'users'
+
+    def absolute_url(self, obj):
+        return obj.absolute_url
+
+    def create(self, validated_data):
+        auth = Auth(self.context['request'].user)
+        node = self.context['view'].get_node()
+        user = get_object_or_404(User, validated_data['_id'])
+        # Node object checks for contributor existence but can still change permissions anyway
+        if user in node.contributors:
+            raise exceptions.ValidationError('{} is already a validated contributor'.format(user.username))
+
+        bibliographic = validated_data['bibliographic']
+        permissions = self.get_permissions_list(validated_data['permission']) \
+            if 'permission' in validated_data else ['read', 'write']
+        node.add_contributor(contributor=user, auth=auth, visible=bibliographic, permissions=permissions, save=True)
+        node.reload()
+        user.permission = node.get_permissions(user)[-1]
```

Comment:
I'm a little confused with what you want me to do here.  I could reload the user object I suppose but nothing would be changed as the node object is the only one being changed.  Contributor permissions and bibliographic status (visibility) are stored in the node object so I'd have to load them anyway.  I decided to load them from the node rather than use the given values to make sure that the user wasn't edited.  The node object does not have a get contributor method that I can find.  Care to clarify some or give advice on what you want me to do?